table/tables/partition.go

```diff
@@ -1073,11 +1073,16 @@ func (t *partitionedTable) GetPartition(pid int64) table.PhysicalTable {
 }
 
 // GetPartitionByRow returns a Table, which is actually a Partition.
-func (t *partitionedTable) GetPartitionByRow(ctx sessionctx.Context, r []types.Datum) (table.PhysicalTable, error) {
+func (t *partitionedTable) GetPartitionByRow(ctx sessionctx.Context, r []types.Datum, tbl interface{}) (table.PhysicalTable, error) {
```

Comment:
How about changing the third argument to `partitionSet *partitionTableWithGivenSets` which specifies some partitions where the returned partition should be in? And there is no limitation if it's nil.

Reply:
I did not want to export the partitionTableWithGivenSets (I.e. rename it to PartitionTableWithGivenSets), but I can do that if you like? (Not sure what our guidelines for exporting types and methods, do we have any document or guidelines for it?)

Reply:
I implemented `partitionTableWithGivenSets.GetPartitionByRow()` and reverted the interface change for `PartitionTable.GetPartitionByRow()`, was is something like that you were proposing?

Reply:
Yes!